### ISSUE TITLE
Curve.add_point has changed and has no compatibility function

### ISSUE THREAD

**Issue by fire** [godotengine/godot-cpp#1512]
### Godot version

https://github.com/godotengine/godot/tree/4ab8fb809396fa38ba929fec97cfcb7193f1c44d

### godot-cpp version

https://github.com/godotengine/godot-cpp/tree/99926d8e2027f6bfdc66e341a1c735b20eda61e4

### System information

Windows 11, Nvidia 3090, Vulkan forward+

### Issue description

We are trying to upgrade to 4.3 and loading with godot double precision crashes when using the gdextension.

https://github.com/Vahera/godot-orchestrator/pull/450

### Steps to reproduce

1. compile godot engine
2. load project

![344434483-7831078d-c271-4f44-a5c0-b04e606e5655](https://github.com/godotengine/godot-cpp/assets/32321/e69aaab9-aee6-46f5-9545-fb6b54b1090c)

### Minimal reproduction project

[godot-orchestrator-demo-windows-release(3).zip](https://github.com/user-attachments/files/16042635/godot-orchestrator-demo-windows-release.3.zip)

**Comment by dsnopek** [comment]
Compatibility methods are something that needs to be added on the Godot side, so this issue should probably be moved there.

Also, you should make sure that you're compiling your GDExtension with `precision=double` and using an `extension_api.json` that was generated from a Godot that was built with double precision. The method hashes can be different between single and double precision builds, so you need to make sure to have those aligned.

**Comment by Naros** [comment]
@dsnopek in that case, I bet its the `extension_api.json` issue then, because I believe Fire and I were using the single precision file that was included in the godot-cpp checkout by default.

**Comment by dsnopek** [comment]
It would be nice to have something in the `extension_api.json` to identify whether it was generated with a single or double precision build, so we could detect and error out if someone is trying to make a `precision=double` build with a bad `extension_api.json`.

I don't know that we should be including the double precision JSON file in godot-cpp, though. Godot doesn't distribute official double precision builds, and so I don't know that we want to include `extension_api.json` files from non-official builds in godot-cpp. If folks are making their own Godot builds (which is required for double precision), they should be using their own `extension_api.json`.

**Comment by fire** [comment]
Yes! An error would be great so I didn’t need to go a wild goose chase.
